### PR TITLE
ARROW-6269: [C++] check decimal precision in IPC code

### PR DESCRIPTION
**Pull request by crepererum**

### CODE REVIEW

File: cpp/src/arrow/ipc/metadata-internal.cc [hunk 1]
```diff
@@ -251,6 +251,9 @@ Status ConcreteTypeFromFlatbuffer(flatbuf::Type type, const void* type_data,
       return Status::OK();
     case flatbuf::Type_Decimal: {
       auto dec_type = static_cast<const flatbuf::Decimal*>(type_data);
+      if ((dec_type->precision() < 1) || (dec_type->precision() > 38)) {
```
Comment (fsaintjacques) [inline cpp/src/arrow/ipc/metadata-internal.cc]: Could we make (1, 38) a constexpr in either type.h or basic_decimal.h. I see them in gandiva's decimal_type_util.h, bring them back in arrow/

**Comment by pitrou** [comment]
Rebased and improved.

**Review by pitrou — APPROVED** [review]
+1, will wait for CI.

**Comment by crepererum** [comment]
Sorry, I was busy since submitting this PR. Thanks @pitrou, looks really good now :)

**Pull request closed without merging**